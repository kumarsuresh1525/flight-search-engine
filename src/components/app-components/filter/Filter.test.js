import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FilterByPrice from './Filter';

describe('', () => {
  const deefaultProps = {
    setFilters: jest.fn(),
    getFlights: jest.fn(),
    filters: {src: 'src'}
  }
  test('renders filter by price correctly', () => {
    const { asFragment } = render(<FilterByPrice {...deefaultProps}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Should handle Price change', () => {
    const { container, getByText } = render(<FilterByPrice {...deefaultProps}/>);
    const input = container.querySelector('.input-range');
    fireEvent.change(input, {target: {value: 2000}});
    const text = getByText(/Price : 2000/);
    expect(text).toBeInTheDocument();
  });
});

