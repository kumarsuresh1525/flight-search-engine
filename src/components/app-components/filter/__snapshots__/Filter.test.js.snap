// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders filter by price correctly 1`] = `
<DocumentFragment>
  <div
    class="price"
  >
    <div
      class="price-label"
    >
      Filter By Price
    </div>
    <div
      class="price-range"
    >
      <span
        class="price-range--min"
      >
        1000
      </span>
      <span
        class="price-range--max"
      >
        12000
      </span>
    </div>
    <div>
      <input
        class="input-range"
        data-testid="input"
        max="12000"
        min="1000"
        type="range"
        value="12000"
      />
    </div>
    <div
      class="price-selected"
    >
      Price : 12000
    </div>
  </div>
</DocumentFragment>
`;
